lib/lotus/entity.rb

```diff
@@ -138,24 +138,35 @@ module ClassMethods
       #
       #   User.attributes => #<Set: {:id, :name}>
       #   DeletedUser.attributes => #<Set: {:id, :name, :deleted_at}>
+      #
       def attributes(*attrs)
         if attrs.any?
           attrs = Lotus::Utils::Kernel.Array(attrs)
           self.attributes.merge attrs
 
           attrs.each do |attr|
-            attr_accessor(attr) if define_attribute?(attr)
+            define_attr_accessor(attr) if defined_attribute?(attr)
```

Comment:
@mvader because it's easier to override than behaviour; when you inject another module.
you can see it there: https://github.com/dsnipe/model/blob/ac9bcd5ac2612371c214177ae7ab4c5c9598b5ee/lib/lotus/entity/dirty_tracking.rb#L45-L53